olx_parser.py

```diff
@@ -0,0 +1,124 @@
+import requests
+from bs4 import BeautifulSoup
+from datetime import datetime, timedelta
+
+
+date = {'январь': 'January',
+        'февраль': 'February',
+        'март': 'March',
+        'апрель': 'April',
+        'май': 'May',
+        'июнь': 'June',
+        'июль': 'July',
+        'август': 'August',
+        'сентябрь': 'September',
+        'октябрь': 'October',
+        'ноябрь': 'November',
+        'декабрь': 'December',
+        'января': 'January',
+        'февраля': 'February',
+        'марта': 'March',
+        'апреля': 'April',
+        'мая': 'May',
+        'июня': 'June',
+        'июля': 'July',
+        'августа': 'August',
+        'сентября': 'September',
+        'октября': 'October',
+        'ноября': 'November',
+        'декабря': 'December'}
+
+
+def get_product(url, count = 100):
+    title_begin = None
+    title_str = None
+    for page in range(1,count):
+        title_begin = title_str
+        try:
+            response = requests.get(f'{url}{page}', timeout=5)
+            response.raise_for_status()
+            soup = BeautifulSoup(response.text, 'lxml')
+            title_str = soup.find('title').text
+            if title_begin == title_str:
+                break
+            data = soup.find_all('div', class_='css-1sw7q4x')
+            for link in data:
+                card_url = link.find('a').get('href')
+                response_product = requests.get('https://www.olx.kz' + card_url, timeout=5)
+                response_product.raise_for_status()                
+                soup_product = BeautifulSoup(response_product.text, 'lxml')
+                soup_product_number_id = soup_product.find('span', class_='css-12hdxwj er34gjf0').text                 
+                soup_product_name_salesman = soup_product.find('h4', class_='css-1lcz6o7 er34gjf0').text
+
+                soup_product_date_registered = soup_product.find('div', class_='css-16h6te1 er34gjf0').text.split(' ')
+                date_registered = format_date(soup_product_date_registered)
+                
+
+                soup_product_date_of_last_visit = soup_product.find('span', class_='css-1t0qnkx').text.split(' ')
+                date_of_last_visit = format_date(soup_product_date_of_last_visit)
+                
+                soup_product_date_posted = soup_product.find('span', class_='css-19yf5ek').text.split(' ')
+                date_posted = format_date(soup_product_date_posted)
+                
+                soup_product_jpg = soup_product.find('img').get('src')                  
+                soup_product_title = soup_product.find('h4', class_='css-1juynto').text
+                soup_product_text = soup_product.find('div', class_='css-1t507yq er34gjf0').text
+        
+        except(requests.RequestException, ValueError, AttributeError):
+            print('error')
+            continue
+
+
+
+def format_date(date_time):
```

Comment:
А пусть функция сама делает split, если уж она отвечает за парсинг даты.
```suggestion
def parse_date_string(date_string: str) -> datetime:
```